File: app/src/main/java/com/zeliot/servicemanager/util/ServiceUtil.java
package com.zeliot.servicemanager.util;

import android.content.ComponentName;
import android.content.Intent;
import android.os.Handler;
import android.os.Looper;

import com.zeliot.servicemanager.model.ServicePackageInfo;

import java.util.List;

public class ServiceUtil {
	private ServiceUtil() {
	}

	public static void startService(
			ServicePackageInfo info) {
		Intent serviceIntent = new Intent();
		serviceIntent.setComponent(
				new ComponentName(info.getContext(), info.getAction()));
		info.getContext().startForegroundService(serviceIntent);
	}

	public static void startMultipleServices(
			List<ServicePackageInfo> packageInfoList) {
		new Handler(Looper.getMainLooper()).postDelayed(() -> {
			for (ServicePackageInfo info : packageInfoList) {
				Intent serviceIntent = new Intent();
				serviceIntent.setComponent(
						new ComponentName(info.getContext(), info.getAction()));
				info.getContext().startForegroundService(serviceIntent);
			}
		}, 100);
	}
}

File: app/src/main/java/com/zeliot/servicemanager/util/ServiceUtil.kt
package com.zeliot.servicemanager.util

import android.content.Intent
import android.os.Handler
import android.os.Looper
import com.zeliot.servicemanager.model.ServicePackageInfo
import java.util.*

object ServiceUtil {
    fun startService(
            info: ServicePackageInfo) {
        val serviceIntent = Intent().apply {
            action = info.action
            `package` = info.packageName
        }

        info.context?.startForegroundService(serviceIntent)
    }

    fun startMultipleServices(
            packageInfoList: ArrayList<ServicePackageInfo>) {
        Handler(Looper.getMainLooper()).post {
            for (info in packageInfoList) {
                val serviceIntent = Intent().apply {
                    action = info.action
                    `package` = info.packageName
                }

                info.context?.startForegroundService(serviceIntent)

                try {
                    Thread.sleep(100)
                } catch (e: InterruptedException) {
                    e.printStackTrace()
                }
            }
        }
    }
}